filter.go

```diff
@@ -82,6 +82,28 @@ func (filter *Filter) Validate() error {
 	if filter.EventFormat != "" && filter.EventFormat != "client" && filter.EventFormat != "federation" {
 		return errors.New("Bad event_format value. Must be one of [\"client\", \"federation\"]")
 	}
+	if result := filter.Room.Timeline.Validate(); result != nil {
+		return result
+	}
+	return nil
+}
+
+func (filter *RoomEventFilter) Validate() error {
+	if filter.Rooms != nil {
+		for _, roomID := range *filter.Rooms {
+			if !isValidRoomID(roomID) {
+				return errors.New("Bad room value. Must be in the form !localpart:domain")
```

Comment:
It would be useful if the error here can return the faulty value, i.e. `return fmt.Errorf("Bad room value %q: must be in form...", roomID)`